src/query/functions/binary/binary.go

```diff
@@ -347,49 +275,34 @@ func (n *binaryNode) processSingleBlock(
 
 		values := step.Values()
 		for _, value := range values {
-			builder.AppendValue(index, returnBoolFn(value))
+			builder.AppendValue(index, fn(value))
 		}
 	}
 
 	return builder.Build(), nil
 }
 
-// If returnBool is false and op is a comparison function,
-// the function should return the scalar value of lhs if the comparison
-// is true; otherwise, it should return 1 if true, and 0 if false
-func funcReturnBool(fn mathFunc, op string, returnBool bool) mathFunc {
-	if returnBool || !isComparison(op) {
-		return fn
-	}
-	return func(left, right float64) float64 {
-		// NB (arnikola): safe to check that this equals 1 since this is
-		// only performed on comparison functions where viable values
-		// are only 1 or 0
-		take := fn(left, right) == 1
-		if take {
-			return left
-		}
-		// NB(arnikola): whereas prometheus does not include not matching
-		// points in the output, we here replace them with NaNs instead
-		return math.NaN()
-	}
-}
-
-func (n *binaryNode) processBothSeries(
+func processBothSeries(
 	lIter, rIter block.StepIter,
```

Comment:
this functions has 6 arguments now. Consider object ? https://softwareengineering.stackexchange.com/questions/145055/are-there-guidelines-on-how-many-parameters-a-function-should-accept